src/VisualStudio/IntegrationTest/IntegrationTests/AbstractEditorTest.cs

```diff
@@ -138,10 +159,52 @@ protected void SendKeys(params object[] keys)
             => Editor.SendKeys(keys);
 
         protected void DisableSuggestionMode()
-            => VisualStudioWorkspaceOutOfProc.SetUseSuggestionMode(false);
+        {
+            VisualStudioWorkspaceOutOfProc.SetUseSuggestionMode(false);
+            WaitForAsyncOperations(FeatureAttribute.Workspace);
+        }
 
         protected void EnableSuggestionMode()
-            => VisualStudioWorkspaceOutOfProc.SetUseSuggestionMode(true);
+        {
+            VisualStudioWorkspaceOutOfProc.SetUseSuggestionMode(true);
+            WaitForAsyncOperations(FeatureAttribute.Workspace);
+        }
+
+        protected void EnableQuickInfo()
```

Comment:
We can.  I don't see how every test ever isn't going to derive from `AbstractEditorTest` at this point, there is too much useful functionality at this point.  Integration tests are naturally going to involve a broad set of functionality.  Sounds like out rules for tests are
- Move all helper methods into `AbstractEditorTest` 
- Use Assert in the test unless it is more than one line, then move that verification to a helper method in `AbstractEditorTest`